### ISSUE TITLE
Bloom Disk Offload Can Not Run Inference

### ISSUE THREAD

**Issue by mallorbc** [huggingface/accelerate#799]
### System Info

```Shell
Package            Version
------------------ ------------
accelerate         0.13.2
anyio              3.6.2
bitsandbytes       0.35.3
certifi            2022.9.24
charset-normalizer 2.1.1
click              8.1.3
commonmark         0.9.1
fastapi            0.85.1
filelock           3.8.0
h11                0.14.0
huggingface-hub    0.10.1
idna               3.4
numpy              1.23.4
nvidia-pyindex     1.0.9
packaging          21.3
Pillow             9.2.0
pip                22.0.4
psutil             5.9.3
pydantic           1.10.2
Pygments           2.13.0
pyparsing          3.0.9
PyYAML             6.0
regex              2022.9.13
requests           2.28.1
rich               12.6.0
setuptools         58.1.0
sniffio            1.3.0
starlette          0.20.4
tokenizers         0.13.1
torch              1.12.1+cu116
torchaudio         0.12.1+cu116
torchvision        0.13.1+cu116
tqdm               4.64.1
transformers       4.23.1
typing_extensions  4.4.0
urllib3            1.26.12
uvicorn            0.19.0
wheel              0.37.1

Python 3.9.15

Running on Ubuntu 20.04 in a docker container.
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

I am trying to run Bloom 176B(I don't care if its slow) on two RTX 3090s, 128GB of RAM, and an NVME SSD.  When I try to load the model, it seems to work with RAM near max usage and the model weights being put to disk when needed.

However, when I try to run inference I get an error that you can see below.  

Here is the code that can be used to reproduce:
```
model = BloomForCausalLM.from_pretrained("bigscience/bloom",device_map="auto",torch_dtype=torch.float16,offload_folder=offload_folder,offload_state_dict=True)

tokenizer = AutoTokenizer.from_pretrained("bigscience/bloom",padding_side="left")

input_tokenized = self.tokenizer(prompt, return_tensors="pt")

gen_tokens = model.generate(input_tokenized["input_ids"].to(0), do_sample=do_sample, temperature=0.7, top_k=50,top_p=0.95,min_length=10, max_length=10)
```

  File "/usr/local/lib/python3.9/site-packages/transformers/generation_utils.py", line 1422, in generate
    return self.sample(
  File "/usr/local/lib/python3.9/site-packages/transformers/generation_utils.py", line 2035, in sample
    outputs = self(
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/accelerate/hooks.py", line 148, in new_forward
    output = old_forward(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/transformers/models/bloom/modeling_bloom.py", line 860, in forward
    transformer_outputs = self.transformer(
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/transformers/models/bloom/modeling_bloom.py", line 745, in forward
    outputs = block(
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/accelerate/hooks.py", line 148, in new_forward
    output = old_forward(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/transformers/models/bloom/modeling_bloom.py", line 431, in forward
    layernorm_output = self.input_layernorm(hidden_states)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/accelerate/hooks.py", line 143, in new_forward
    args, kwargs = module._hf_hook.pre_forward(module, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/accelerate/hooks.py", line 250, in pre_forward
    set_module_tensor_to_device(module, name, self.execution_device, value=self.weights_map[name])
  File "/usr/local/lib/python3.9/site-packages/accelerate/utils/modeling.py", line 124, in set_module_tensor_to_device
    new_value = value.to(device)
NotImplementedError: Cannot copy out of meta tensor; no data!

### Expected behavior

```Shell
I expect a generated output, regardless of how slow it is.
```

**Comment by mallorbc** [comment]
I will test this on a system that has enough RAM and VRAM to not use the disk.  Maybe that is the issue.  I will also hopefully be able to get an 8XA100 system and test with the model just in VRAM.

**Comment by mallorbc** [comment]
I could get it working 4 A100s(could use 3) with this code after installing bitsandbytes:
```
mem_map = {}
num_gpus = torch.cuda.device_count()
        if num_gpus <=4:    
                max_string = "60GB"
                for i in range(num_gpus):
                    mem_map[i] = max_string
                model = BloomForCausalLM.from_pretrained(self.model_name,device_map="auto",load_in_8bit=True,max_memory=mem_map)
```
I can get the model loaded with this code and if I have more than 4 GPUs I would assume it would work but it errors out when I try inference:
```
  model = BloomForCausalLM.from_pretrained(self.model_name,device_map="auto")
```

Since I have used CPU offload before for other models, I assume this is a disk offload issue.

**Comment by sgugger** [comment]
I think this particular bug might be fixed by #795. I could run BLOOM with offload to disk on a small machine after it.
